### PR TITLE
Add tox config

### PR DESCRIPTION
**Pull request by l00ptr**
We add some basic config for tox and also adapt the CI to use that new config.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -15,7 +15,8 @@ ignore =
   E704,
   # line break before binary operator
   W503,
-
+exclude =
+  .tox
 [tool:pytest]
```
Comment (dlax) [inline setup.cfg]: Can you keep a blank line above the section heading?

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,32 @@
+[tox]
+minversion = 4.20.0
+envlist = lint,tests,typing
+isolated_build = true
+
+[testenv]
+skip_install = true
```
Comment (dlax) [inline tox.ini:7]: IMO, this should not be the default, but rather be set in environments where it's actually needed. Accordingly, `-e .` entries in `deps` below should be removed.

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+[testenv:lint]
+commands=
+  flake8 .
+  black --check --diff .
+  check-manifest .
+  isort --check --diff .
```
Comment (dlax) [inline tox.ini:14]: Do we need the `.`s?

**Review by dlax — CHANGES_REQUESTED** [review]
Thank you, that seems nice to have. LGTM, just a few nits in `tox.ini`.

File: tox.ini (hunk 3 above)
Comment (l00ptr) [reply tox.ini]: it seems like we need it for `black` and `isort`, maybe we can define the SRC by adding some specific config files.

``` console
$ black --check 
Usage: black [OPTIONS] SRC ...

One of 'SRC' or 'code' is required.
```

File: tox.ini (hunk 2 above)
Comment (dlax) [reply tox.ini]: It seems to me that some environments, `typing` and `lint`, can still have `skip_install = true`.

File: tox.ini (hunk 3 above)
Comment (dlax) [reply tox.ini]: Indeed; let's keep those then.

**Review by dlax — APPROVED** [review]

**Pull request merged**